src/kdalgorithms.h

```diff
@@ -382,6 +382,44 @@ ReturnType accumulate_if(const Container &container, BinaryOperation &&accumulat
 
     return kdalgorithms::accumulate(container, fn, initialValue);
 }
+
+// -------------------- sum --------------------
+template <typename Container, typename Projection,
+          typename ReturnType = remove_cvref_t<detail::return_type_of_t<Projection>>>
```

Comment:
OTOH, something like this would be more general:
std::result_of_t/invoke_result_t<Projection, typename Container::value_type>

Reply:
Done.